lib/lints/avoid_late_keyword/avoid_late_keyword_rule.dart

```diff
@@ -31,8 +33,14 @@ class AvoidLateKeywordRule extends SolidLintRule {
     CustomLintContext context,
   ) {
     context.registry.addVariableDeclaration((node) {
-      if (node.declaredElement?.isLate ?? false) {
+      final initializedDynamically = config.parameters.initializedDynamically;
+      final isLateDeclaration = node.declaredElement?.isLate ?? false;
+      final hasInitializer = node.initializer != null;
+
+      if ((!initializedDynamically || !hasInitializer) && isLateDeclaration) {
         reporter.reportErrorForNode(code, node);
+      } else if (initializedDynamically && hasInitializer) {
+        return;
```

Comment:
Second check doesn't do anything useful.

```suggestion
      if ((!initializedDynamically || !hasInitializer) && isLateDeclaration) {
        reporter.reportErrorForNode(code, node);
```